Index: date_api_sql.inc
===================================================================
--- date_api_sql.inc	(revision 18762)
+++ date_api_sql.inc	(working copy)
@@ -213,7 +213,7 @@
             return "ADDTIME($field, SEC_TO_TIME($offset))";
           }
           else {
-            return "DATE_ADD($field, INTERVAL $offset SECOND)";
+            return "DATE_ADD(CAST($field AS DATETIME), INTERVAL $offset SECOND)";
           }
         case 'pgsql':
           return "($field + INTERVAL '$offset SECONDS')";;
@@ -242,9 +242,9 @@
       case 'mysqli':
         switch ($direction) {
           case 'ADD':
-            return "DATE_ADD($field, INTERVAL $count $granularity)";
+            return "DATE_ADD(CAST($field AS DATETIME), INTERVAL $count $granularity)";
           case 'SUB':
-            return "DATE_SUB($field, INTERVAL $count $granularity)";
+            return "DATE_SUB(CAST($field AS DATETIME), INTERVAL $count $granularity)";
           }
 
       case 'pgsql':
